### PR TITLE
feat: add exec_cell_with_args

### PR DESCRIPTION
**Pull request by quake**
The `high_level::exec_cell` fn requires the user to build `CStr`, but in most of the scenarios, the args are passed as u8 slice, the callee will parse the args by itself, building `CStr` also needs to deal with the problem of \0 in slice, this method is not convenient in practice, it does not feel like **high_level**. 

This PR provides a new fn to pass u8 slice directly:

caller
```
let args = [42, 0, 0, 0, 42, 0];
high_level::exec_cell_with_args(code_hash, hash_type, 0, 0, &args)
```

callee
```
let args = ckb_std::env::args();
assert_eq!(&[42, 0, 0, 0, 42, 0], args.as_slice());
```

### CODE REVIEW

**Comment by XuJiandong** [comment]
Suggestion:
&CStr, CString, String or &Str. 
The usage slice is dangerous if tailing zero is not appended.

**Comment by quake** [comment]
> Suggestion: &CStr, CString, String or &Str. The usage slice is dangerous if tailing zero is not appended.

no need to append tailing zero, we have handled it in `exec_cell_with_args`, code: https://github.com/nervosnetwork/ckb-std/blob/ca4dddc90f47d6c88399e42398019b24b436ddc9/src/high_level.rs#L641-L644

user can pass slice directly:

```
let args = [1, 0, 1, 0, 1, 2, 3, 4];
high_level::exec_cell_with_args(code_hash, hash_type, 0, 0, &args)
```

```
let args = ckb_std::env::args();
assert_eq!(&[1, 0, 1, 0, 1, 2, 3, 4], args.as_slice());
```

**Comment by XuJiandong** [comment]
CStr, CString are different than slice:
1. UTF-8 awareness
2. no null bytes in the middle

Using slice is C-style and not recommended.

File: src/high_level.rs [hunk 1]
```diff
@@ -622,6 +622,47 @@ pub fn exec_cell(
...
+/// * `hash_type` - the hash type to search cell in cell deps.
+/// * `offset`    - the offset of the cell data to load.
+/// * `length`    - the length of the cell data to load, if the length is 0, load the whole cell data starting from offset.
+/// * `args`      - the arguments to pass to the execution.
+pub fn exec_cell_with_args(
+    code_hash: &[u8],
+    hash_type: ScriptHashType,
+    offset: u32,
+    length: u32,
+    args: &[u8],
```
Comment (blckngm) [inline src/high_level.rs:639]: This is very different from the RFC or how args are passed in C/POSIX. We need to document the exact mechanism and how to recover it on the callee side if we are really going to add this.

**Comment by blckngm** [comment]
Conventionally the args are utf-8 `CStr`s, not an arbitrary binary slice.

File: src/high_level.rs [hunk 2]
```diff
@@ -622,6 +622,47 @@ pub fn exec_cell(
...
+) -> Result<Infallible, SysError> {
+    let args_with_nul: Vec<Vec<u8>> = args
+        .split(|i| *i == 0)
+        .map(|slice| [slice, &[0]].concat().to_vec())
```
Comment (blckngm) [inline src/high_level.rs:643]: Should be possible to optimize this to only allocate and copy for the last slice.

**Pull request closed without merging**